src/lib/utils/functions/FileHandleUtils.ts

```diff
@@ -1,24 +1,78 @@
 import {
+  EntityBundle,
+  EntityType,
+  EXTERNAL_FILE_HANDLE_CONCRETE_TYPE_VALUE,
+  EXTERNAL_OBJECT_STORE_FILE_HANDLE_CONCRETE_TYPE_VALUE,
   ExternalFileHandle,
   ExternalObjectStoreFileHandle,
+  FileEntity,
   FileHandle,
-  ProxyFileHandle,
+  GOOGLE_CLOUD_FILE_HANDLE_CONCRETE_TYPE_VALUE,
   GoogleCloudFileHandle,
-  S3FileHandle,
   PROXY_FILE_HANDLE_CONCRETE_TYPE_VALUE,
-  EXTERNAL_OBJECT_STORE_FILE_HANDLE_CONCRETE_TYPE_VALUE,
-  EXTERNAL_FILE_HANDLE_CONCRETE_TYPE_VALUE,
+  ProxyFileHandle,
   S3_FILE_HANDLE_CONCRETE_TYPE_VALUE,
-  GOOGLE_CLOUD_FILE_HANDLE_CONCRETE_TYPE_VALUE,
+  S3FileHandle,
 } from '../synapseTypes'
+import {
+  ExternalGoogleCloudUploadDestination,
+  ExternalObjectStoreUploadDestination,
+  ExternalS3UploadDestination,
+  ExternalUploadDestination,
+  UploadDestination,
+  UploadType,
+} from '../synapseTypes/File/UploadDestination'
+
+const SYNAPSE_STORAGE = 'Synapse Storage'
+const SYNAPSE_STORAGE_LOCATION_ID = 1
+
+type FileHandleStorageInfo =
+  | {
+      endpoint: string
+      bucket: string
+      fileKey: string
+    }
+  | { url: string }
+  | { location: string }
```

Comment:
On the entity page, this is how we currently break down and report different file handle types.